### PR TITLE
transpose exercise

### PR DESCRIPTION
**Pull request by Ephraim-nonso**
Closes #ISSUE_ID

### CODE REVIEW

File: config.json [hunk 1]
```diff
@@ -820,6 +820,14 @@
         "prerequisites": [],
         "difficulty": 4
       },
+      {
+        "slug": "word-count",
```
Comment (0xNeshi) [inline config.json:824]: You already added `word-count`, see https://github.com/exercism/cairo/pull/304/files
Delete these files from the PR.

**Review by 0xNeshi — CHANGES_REQUESTED** [review]

File: config.json (hunk 1 above)
Comment (Ephraim-nonso) [reply config.json:824]: Apologies for this error. I had issues which prompted this error along the line. I'd delete it from the PR.

Comment (Ephraim-nonso) [reply config.json:824]: I have deleted these files from the PR. Please, let me know if there are other changes to make.

Comment (0xNeshi) [reply config.json:824]: No problem at all, it happens!

Comment (0xNeshi) [reply config.json:824]: The files are still in the PR :sweat_smile:

File: exercises/practice/transpose/tests/transpose.cairo [hunk 2]
```diff
@@ -0,0 +1,174 @@
+use transpose::transpose as transpose_matrix;
+
+#[test]
+fn empty_string() {
+    let mut input: Array<ByteArray> = array![];
```
Comment (0xNeshi) [inline exercises/practice/transpose/tests/transpose.cairo:5]:
```suggestion
    let input: Array<ByteArray> = array![];
```
no need for `input` to be mutable, address this in all test cases

File: exercises/practice/transpose/tests/transpose.cairo [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+
+    let output = transpose_matrix(input);
+    assert_eq!(output, expected);
+}
+
+#[test]
+#[ignore]
+fn simple() {
+    let mut input: Array<ByteArray> = array!["ABC", "123"];
+    let expected = array!["A1", "B2", "C3"];
```
Comment (0xNeshi) [inline exercises/practice/transpose/tests/transpose.cairo:36]: nit: let's format these row values into _rows_, and column values into _columns_. 

You can use `#[cairofmt::skip]` to tell the formatter to not touch this then.
This is similar to what we've done for [minesweeper](https://github.com/exercism/cairo/blob/01caba201497103feb5bdc6b83995420b3672a42/exercises/practice/minesweeper/tests/minesweeper.cairo#L57).

```rust
#[cairofmt::skip]
let input: Array<ByteArray> = array![
    "ABC", 
    "123",
];
#[cairofmt::skip]
let expected = array![
    "A1", 
    "B2",
    "C3",
];
```

Note: apply this change to all tests where the formatter didn't already format the values like this

File: exercises/practice/transpose/tests/transpose.cairo [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+    ];
+    let expected = array![
+        "TAAA",
+        "h   ",
+        "elll",
+        " ooi",
+        "lnnn",
+        "ogge",
+        "n e.",
+        "glr ",
```
Comment (0xNeshi) [inline exercises/practice/transpose/tests/transpose.cairo:120]: You should not pad to the right, i.e. add padding spaces to the rightmost columns (as per [the problem description](https://github.com/exercism/problem-specifications/blob/main/exercises/transpose/description.md?plain=1#L26)).

Copy/paste the actual expected output from [the canonical data](https://github.com/exercism/problem-specifications/blob/main/exercises/transpose/canonical-data.json#L156)

File: exercises/practice/transpose/tests/transpose.cairo [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+
+    let output = transpose_matrix(input);
+    assert_eq!(output, expected);
+}
+
+#[test]
+#[ignore]
+fn jagged_triangle() {
+    let mut input: Array<ByteArray> = array!["11", "2", "3333", "444", "555555", "66666"];
+    let expected = array!["123456", "1 3456", "  3456", "  3 56", "    56", "    5 "];
```
Comment (0xNeshi) [inline exercises/practice/transpose/tests/transpose.cairo:170]:
```suggestion
    let expected = array!["123456", "1 3456", "  3456", "  3 56", "    56", "    5"];
```
Same as before, no padding to the right is allowed (see [canonical data](https://github.com/exercism/problem-specifications/blob/main/exercises/transpose/canonical-data.json#L263))

File: exercises/practice/transpose/.meta/example.cairo [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+    for line in input.clone() {
+        if line.len() > max_length {
+            max_length = line.len();
+        }
+    };
+
+    let mut i = 0;
+    loop {
+        let mut temp: ByteArray = "";
+        for line in input
```
Comment (0xNeshi) [inline exercises/practice/transpose/.meta/example.cairo:14]: It must be possible to avoid having a `loop + for + break + continue` combination to perform this operation, makes it very hard to follow what is going on.
 
Find a way to refactor this into something more readable.

You can use [Go implementation](https://github.com/exercism/go/blob/main/exercises/practice/transpose/.meta/example.go) as inspiration, it is very clear and concise

File: config.json [hunk 7]
```diff
@@ -852,6 +860,14 @@
         "prerequisites": [],
         "difficulty": 6
       },
+      {
+        "slug": "transpose",
+        "name": "Transpose",
+        "uuid": "59b5d30b-62c1-4465-a44d-485c45f4aac2",
+        "practices": [],
+        "prerequisites": [],
+        "difficulty": 1
```
Comment (0xNeshi) [inline config.json:869]:
```suggestion
        "difficulty": 4
```
It's not a trivial exercise, let's increase the difficulty

**Review by 0xNeshi — CHANGES_REQUESTED** [review]
Good first iteration, let's polish it a bit!

**Comment by Ephraim-nonso** [comment]
> Good first iteration, let's polish it a bit!

Thanks for the review. I'd address them.

**Comment by Ephraim-nonso** [comment]
@0xNeshi

**Comment by 0xNeshi** [comment]
Hey @Ephraim-nonso , looks like there are still some unaddressed comments from the previous review - could you take a look at these points here: https://github.com/exercism/cairo/pull/315#pullrequestreview-2494510040?
Once those are sorted out we can move forward with the next review.
Thanks!

**Comment by Ephraim-nonso** [comment]
I'd attend to the other comments and fix others. Sorry for the delay.

**Comment by 0xNeshi** [comment]
Hey @Ephraim-nonso, hope all is well! 

Your work on this PR, and on this track as a whole, has been amazing so far - I really wanted to commend you on that! This PR is so close to being merged and will add a completely new exercise for all Cairo students, which is fantastic.

I wanted to check in and see how things are going with the remaining work. Is there anything blocking you or any way I can help move this forward? No rush at all - just wanted to make sure you have what you need to finish strong.

Thanks again for all your contributions!
